nexus/db-queries/src/authz/api_resources.rs

```diff
@@ -867,6 +867,23 @@ authz_resource! {
     polar_snippet = Custom,
 }
 
+authz_resource! {
+    name = "SiloImage",
+    parent = "Silo",
+    primary_key = Uuid,
+    roles_allowed = false,
+    polar_snippet = InSilo,
+}
+
+// This resource is a collection of _all_ images in a silo, including project images.
+authz_resource! {
+    name = "Image",
+    parent = "Silo",
+    primary_key = Uuid,
+    roles_allowed = false,
+    polar_snippet = InSilo,
+}
```

Comment:
It's notable here that I've used the `InSilo` polar snippet to define auth for the base image when that's not _strictly_ true. The only place `Image` is referred to in the auth system is when we're looking up an image based on ID. In that case, we don't know if it's a project or silo image so we must use the generic form of the image lookup. For the current implementation that means a user _must_ have read permissions on the Silo to even be able to look up a project image by ID. Subsequent authz checks validates the appropriate permissions on the project (if it's a project image) so we get the right checks on the resource eventually.